core/channel.go

```diff
@@ -74,6 +81,55 @@ func CreateChannel(pathName string, src, dst *ProvableChain, to time.Duration) e
 	return nil
 }
 
+func checkChannelCreateReady(src, dst *ProvableChain) (bool, error) {
+	srcID := src.Chain.Path().ChannelID
+	dstID := dst.Chain.Path().ChannelID
+
+	if srcID == "" && dstID == "" {
+		return true, nil
+	}
+
+	getState := func(pc *ProvableChain) (chantypes.State, error) {
+		if pc.Chain.Path().ChannelID == "" {
+			return chantypes.UNINITIALIZED, nil
+		}
+
+		latestHeight, err := pc.LatestHeight()
+		if err != nil {
+			return chantypes.UNINITIALIZED, err
+		}
+		queryHeight := clienttypes.NewHeight(latestHeight.GetRevisionNumber(), 0)
```

Comment:
In most cases, `RevisionHeight == 0` means the genesis height.
In this case you should simply use the latest height.

Reply:
fixed